src/test/java/org/mastodon/mamut/feature/spot/ellipsoid/SpotEllipsoidFeatureTest.java

```diff
@@ -0,0 +1,102 @@
+package org.mastodon.mamut.feature.spot.ellipsoid;
+
+import org.junit.Test;
+import org.mastodon.feature.Feature;
+import org.mastodon.feature.FeatureProjectionKey;
+import org.mastodon.mamut.feature.FeatureSerializerTestUtils;
+import org.mastodon.mamut.feature.branch.FeatureComputerTestUtils;
+import org.mastodon.mamut.feature.exampleGraph.ExampleGraph3;
+import org.mastodon.mamut.model.Model;
+import org.mastodon.mamut.model.Spot;
+import org.scijava.Context;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SpotEllipsoidFeatureTest
+{
+
+	@Test
+	public void testEllipsoidFeature()
+	{
+		try (Context context = new Context())
+		{
+			ExampleGraph3 exampleGraph3 = new ExampleGraph3();
+
+			// set an example covariance matrix
+			double[][] covarianceMatrix = new double[][] { { 6, 2, 3 }, { 2, 7, 4 }, { 3, 4, 8 } };
+			exampleGraph3.spot0.setCovariance( covarianceMatrix );
+
+			// compute feature
+			Feature< Spot > ellipsoidFeature =
+					FeatureComputerTestUtils.getSpotFeature( context, exampleGraph3, SpotEllipsoidFeature.SPOT_ELLIPSOID_FEATURE_SPEC );
+
+			// eigenvalues for given covariance matrix
+			// cf. https://matrixcalc.org/de/vectors.html#eigenvectors({{6, 2, 3}, {2, 7, 4}, {3, 4, 8}})
+			final double[] eigenValues = new double[] { 3.270d, 4.442d, 13.288d };
+
+			// compute semi-axes from eigenvalues
+			double expectedShortAxis = Math.sqrt( eigenValues[ 0 ] );
+			double expectedMiddleAxis = Math.sqrt( eigenValues[ 1 ] );
+			double expectedLongAxis = Math.sqrt( eigenValues[ 2 ] );
+			double expectedVolume = expectedShortAxis * expectedMiddleAxis * expectedLongAxis * 4d / 3d * Math.PI;
+
+			// check that the features are computed correctly
+			assertEquals( expectedShortAxis,
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.SHORT_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedMiddleAxis,
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.MIDDLE_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			assertEquals( expectedLongAxis,
+					ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.LONG_SEMI_AXIS_PROJECTION_SPEC ) )
+							.value( exampleGraph3.spot0 ),
+					0.001d );
+			// volume of ellipsoid https://en.wikipedia.org/wiki/Ellipsoid#Volume
+			assertEquals( expectedVolume, ellipsoidFeature
+					.project( FeatureProjectionKey.key( SpotEllipsoidFeature.VOLUME_PROJECTION_SPEC ) ).value( exampleGraph3.spot0 ),
+					0.01d );
```

Comment:
These lines are too long and awfully hard to read. Make it simpler:
```java
SomeFeatureProjectionType shortProjection = ellipsoidFeature.project( FeatureProjectionKey.key( SpotEllipsoidFeature.SHORT_SEMI_AXIS_PROJECTION_SPEC ) );
assertEquals( expectedShortAxis,  shortProjection.value( exampleGraph3.spot0 ), 0.001d );
```

Reply:
I changed it according to your advice.

Before, I intentionally did this not, because with this advice variables are introduced that are only used once. This adds extra lines of code that are actually not needed. I guess, this is where clean (readable) code becomes subjective as in the past, I was also criticized (also referring to clean code) to not introduce variables that are only used once (as it adds unnecessary lines of code).

Reply:
Clean code is subjective definitely. But there is no such rule to remove every variable that's only used once. There was either a misunderstanding in the past or you got pretty bad advice.